### ISSUE TITLE
BUG: SlicePlot breaks if image only contains zeros and invalid values

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#3826]
### Bug report

**Bug summary**
It's currently impossible to produce a SlicePlot of a field full of zeros if it's projection doesn't fill the image.
This is easy to demonstrate with a non-cartesian dataset, here using a `"polar"` geometry

**Code for reproduction**

```python
import numpy as np
import yt
from yt.units import g, cm

shape = (16, 16, 16)
data = {"density": np.zeros(shape) * g/cm**3}

ds = yt.load_uniform_grid(
    data,
    shape,
    bbox=np.array([[0, 1], [0, 1], [0, 1]]),
    geometry="polar",
)

p = yt.SlicePlot(ds, "z", ("gas", "density"))
p.save("/tmp/")
```

**Actual outcome**

```
Traceback (most recent call last):
  File "t_zero_field.py", line 16, in <module>
    p = yt.SlicePlot(ds, "z", ("gas", "density"))
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/plot_window.py", line 1971, in __init__
    PWViewerMPL.__init__(
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/plot_window.py", line 851, in __init__
    PlotWindow.__init__(self, *args, **kwargs)
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/plot_window.py", line 251, in __init__
    self._setup_plots()
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/plot_window.py", line 1118, in _setup_plots
    self.plots[f] = WindowPlotMPL(
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/plot_window.py", line 2636, in __init__
    self._init_image(data, cbname, cblinthresh, cmap, extent, aspect)
  File "/Users/robcleme/dev/yt-project/yt/yt/visualization/base_plot_types.py", line 343, in _init_image
    self.cb = self.figure.colorbar(self.image, self.cax)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/figure.py", line 1158, in colorbar
    cb = cbar.Colorbar(cax, mappable, **cb_kw)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 484, in __init__
    self._reset_locator_formatter_scale()
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 1175, in _reset_locator_formatter_scale
    self._process_values()
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 1113, in _process_values
    self.norm.vmin, self.norm.vmax = mtransforms.nonsingular(
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colors.py", line 1150, in vmin
    self._changed()
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colors.py", line 1178, in _changed
    self.callbacks.process('changed')
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 292, in process
    self.exception_handler(exc)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 96, in _exception_printer
    raise exc
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/image.py", line 308, in changed
    cm.ScalarMappable.changed(self)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cm.py", line 618, in changed
    self.callbacks.process('changed', self)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 292, in process
    self.exception_handler(exc)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 96, in _exception_printer
    raise exc
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 547, in update_normal
    self.draw_all()
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 574, in draw_all
    self._process_values()
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colorbar.py", line 1117, in _process_values
    b = self.norm.inverse(b)
  File "/Users/robcleme/.pyenv/versions/yt-dev/lib/python3.8/site-packages/matplotlib/colors.py", line 1554, in inverse
    raise ValueError("Invalid vmin or vmax")
ValueError: Invalid vmin or vmax
```

**Expected outcome**

A nice image.
Note that the script above works if we use a cartesian geometry instead.
![UniformGridData_Slice_z_density](https://user-images.githubusercontent.com/14075922/155557437-f3f1bb74-fc0e-48aa-88b2-1d8435dca7b2.png)

It's possible that the actual bug is in matplotlib. I'll investigate this further.

**Comment by neutrinoceros** [comment]
essentially the scenario can be reproduced in pure mpl:
```python
import matplotlib.pyplot as plt
from matplotlib.colors import LogNorm
import numpy as np

shape = (20, 10)
data = np.concatenate([np.zeros(shape), np.full(shape, np.nan)])
fig, ax = plt.subplots()
image = ax.imshow(data, norm=LogNorm())
fig.colorbar(image)
```


However at this point I don't think there's anything matplotlib can do about it, so I'm tempted to conclude we need to fix this on our side by switching to a linear norm (consistently with what is done in the cartesian case).

**Comment by neutrinoceros** [comment]
Turns out I already fixed this with #3793, I just didn't synchronise my main branch since it was merged !
